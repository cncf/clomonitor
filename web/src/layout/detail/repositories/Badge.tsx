import { isUndefined } from 'lodash';
import { useLocation, useNavigate } from 'react-router-dom';

import getCategoryColor from '../../../utils/getCategoryColor';
import styles from './Badge.module.css';

interface Props {
  value?: number;
  linkTo: string;
  scrollIntoView: (id?: string) => void;
}

const Badge = (props: Props) => {
  const navigate = useNavigate();
  const location = useLocation();

  if (isUndefined(props.value))
    return (
      <div className="mx-auto px-2 text-center">
        <span className="text-muted">n/a</span>
      </div>
    );

  const color = getCategoryColor(props.value);

  return (
    <div
      className={`mx-auto px-2 text-center ${styles.badge}`}
      style={{
        borderBottomColor: `var(--rm-${color})`,
      }}
    >
      <button
        className={`btn btn-link text-dark fw-bold p-0 text-decoration-none ${styles.btn}`}
        onClick={() => {
          props.scrollIntoView(`#${props.linkTo}`);
          navigate(
            {
              pathname: location.pathname,
              hash: props.linkTo,
            },
            { state: location.state }
          );
        }}
        aria-label={`Go from summary to section: ${props.linkTo}`}
      >
        {props.value}
      </button>
    </div>
  );
};

export default Badge;
