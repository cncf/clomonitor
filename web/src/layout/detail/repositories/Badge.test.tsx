import { render, screen } from '@testing-library/react';

import Badge from './Badge';

describe('Badge', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<Badge value={80} />);

    expect(asFragment()).toMatchSnapshot();
  });

  describe('Render', () => {
    it('renders badge', () => {
      render(<Badge value={80} />);

      expect(screen.getByText('80')).toBeInTheDocument();
    });

    it('renders badge with undefined value', () => {
      render(<Badge />);

      expect(screen.getByText('n/a')).toBeInTheDocument();
    });
  });
});
