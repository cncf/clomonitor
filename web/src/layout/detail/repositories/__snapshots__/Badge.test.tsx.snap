// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="mx-auto px-2 text-center badge"
  >
    <span
      class="text-dark fw-bold"
    >
      80
    </span>
  </div>
</DocumentFragment>
`;
